### PR TITLE
Create void script.

### PR DESCRIPTION
**Pull request by ericonr**
This will be shipped in the void-docs package, and can be used to
retrieve documentation pages.

Looking for:

- better name
- command line flags that could be passed
- ...

Thanks @travankor for the suggestion!

### CODE REVIEW

**Comment by ericonr** [comment]
Current ideas:

- fzf selector when multiple results are found
- parametrization of the script for different DESTDIRS (?)
- some way to launch into the summary / main page

**Comment by ericonr** [comment]
I need some UX evaluation :p 

I think the program is looking fine, but I just coded it so I am probably quite biased.

Also, how should it behave when it can't find `fzf` but finds several results for the search? Should I have a command line option to force it to show search results instead of opening the page in such situations? I'm pending towards that.

**Comment by ericonr** [comment]
I guess I should switch the void-docs manpage to section 1, now that it's a program?

**Comment by ericonr** [comment]
@flexibeast I would like to merge this as soon as possible so we can have it on our systems by default and try to spot any issues with it. Otherwise it will only get very limited testing.

**Comment by flexibeast** [comment]
There seem to be significant display errors when using `elinks`:

pic-4

The character doubling appears as one moves down through the list of entries.

**Comment by flexibeast** [comment]
Hmm, it seems to be an `elinks` issue, as i have the same problem when using it to visit https://docs.voidlinux.org/. However, both `links` and `lynx` work satisfactorily (although `links` doesn't highlight the first character of an entry).

**Comment by ericonr** [comment]
Thanks for testing! I will leave `lynx` as the first one in the cli browsers list, and remove `elinks`.

Having seen that comment on reddit where you tagged me, I'm considering having some functionality similar to what `xlocate` does, allowing a user to ever so often download a local version of the docs. But I will make an issue about it once this is merged, so we can properly discuss how to do it and if it's worth doing.

File: void-docs.in [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# knows how to open a certain page
+open_page() {
+    if [ "$_html" ]; then
+        # check if wayland or X session
+        if [ -n "$WAYLAND_DISPLAY" -o -n "$DISPLAY" ] && command -v xdg-open >/dev/null; then
+            page=$(find_page html $1)
+            exec xdg-open $page
+        fi
+
+        if [ -n "$BROWSER" ] && command -v $BROWSER >/dev/null; then
```
Comment (sgn) [inline void-docs.in:79]: I think `$BROWSER` should be given more priority than `xdg-open`.
`xdg-open` could be pulled by some packages,
`$BROWSER` is usually set explicitly by users.

File: void-docs.in [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            page=$(find_page html $1)
+            exec $BROWSER $page
+        fi
+
+        # if a browser or xdg-open weren't available, we can try other stuff
+
+        # cli browser
+        for viewer in lynx links
```
Comment (sgn) [inline void-docs.in:87]: `w3m` could be a good candidate, (I haven't tested, though).

File: void-docs.in (hunk 1 above)
Comment (ericonr) [reply void-docs.in:79]: Good idea, thanks.

File: void-docs.in (hunk 2 above)
Comment (ericonr) [reply void-docs.in:87]: Hmm it doesn't work great, but it's ok. Thanks for the suggestion!

**Comment by ericonr** [comment]
TODO: rename the man page in the README, since it's now `.1`

**Comment by ericonr** [comment]
@flexibeast @bobertlo can I merge?

**Comment by flexibeast** [comment]
@ericonr: i'll leave it to @bobertlo to make any comments on the script itself, but a couple of issues i noticed with `void-docs.md`:

* there are two EXAMPLES sections. There should only be one, after the FILES section.
* i assume the EXAMPLES header should only have one `#`, to be at the same level as other sections. Unless one needs to do that for the conversion for some reason?

**Comment by ericonr** [comment]
So, I tried having an examples section for the `void-docs` command and one for the files in the package. The generated man page seemed to have them located and indented correctly. Is that too non-standard or going to bring issues to the HTML generation? If not, I would like to keep them.

**Comment by flexibeast** [comment]
i have no idea whether it would cause issues for the HTML generation, but i think that would be worth checking - we've already seen how much trouble was caused by the `xtools` man page not following the format expected by `mandoc`. My reading of `mdoc(7)` is that EXAMPLES should be a section, not a subsection as currently generated. 

More generally, i've just checked the `void-docs.1` is generated by pandoc, and it's .... not good. For example, there are no initial `Dd` and `Dt` lines, and flags aren't indicated by use of the `Fl` macro. Again, this might cause issues when mandoc converts to HTML. i'd need to recreate my docs.voidlinux.org setup to check (some of which i lost when i changed from runit to 66).

**Comment by flexibeast** [comment]
So here's how the page looks (the qutebrowser example isn't actually doubled, i just selected a larger region for the final screenshot than i should have):

pic-6
pic-7
pic-8
pic-9

So it's overall usable (although there's excessive vertical whitespace).

Still, given what i've seen of pandoc's roff output, my preference would be that we not use it for generating man pages, and that we manually create such pages instead (which i'd be happy to do). But since pandoc's output is usable, i don't think this issue needs to be a blocker.

**Comment by ericonr** [comment]
Fair enough. If you'd like to perform the conversion and push it onto this PR, I'm also ok with it. There are softwares for proper man page conversion, such as `asciidoc` or `scdoc`, which we could also try to use.

**Comment by ericonr** [comment]
Update to mention PDF version, and a bunch of miscellaneous fixes.

**Comment by ericonr** [comment]
@flexibeast I templated the `@PREFIX@` in the man page as well for consistency, what do you think?

**Comment by flexibeast** [comment]
@ericonr: Seems like a good idea - thanks!

**Comment by ericonr** [comment]
@flexibeast I did some re-organizing, since the repo root was getting a bit crowded there. I would like to have the script from #442 in `res/` as well, what do you think?

**Comment by flexibeast** [comment]
@ericonr: Reorg is definitely a good idea - thanks! i'll update #442 accordingly. 'res[ources]' seems like an odd name for a directory containing several executables in addition to non-executables, but i'm not going to bikeshed over this. :-)

**Comment by ericonr** [comment]
If you feel like it's mergeable, go ahead! Then you can rebase yours :)

**Comment by flexibeast** [comment]
Unfortunately, moving `mdbook-latex` into `res/` results in broken internal links in the PDF, because (i believe) the paths to files in `src/` are now different relative to the `mdbook-latex` script. :-(

Since this PR looks okay to me otherwise, my thought is to merge it, and i can then fix `mdbook-latex` in a separate PR. WDYT?

**Comment by ericonr** [comment]
Hmm did you run the build.sh script as `res/build.sh`?

**Comment by flexibeast** [comment]
i did indeed, and the build completely successfully. But when an internal link doesn't point to a usable hypertarget, `pdflatex` doesn't fail, it just quietly resolves those links to a 'dummy' hypertarget. Can you confirm that if you build the PDF, and try to visit an internal link in the body of the document, you're just taken to page 1?

(Getting internal links working was perhaps the most complex and tricky part of putting together the `mdbook-latex` script.)

**Comment by ericonr** [comment]
I tried deleting `book/latex` and then running `res/build.sh`. The generated PDF is attached, and it seems to work without any issues.
[handbook.pdf](https://github.com/void-linux/void-docs/files/5019586/handbook.pdf)

**Comment by ericonr** [comment]
We should try to investigate this lack of reproducibility... Have you tried cloning my branch directly? I remember this worked when I was trying out #442

**Comment by flexibeast** [comment]
Yes, i did a fresh clone of your repo, and checked out your `bro` branch, to test this.

Does it also work for you if you do a fresh clone?

**Comment by ericonr** [comment]
A fresh clone is also working for me.

**Comment by flexibeast** [comment]
:-/ i really don't understand what's going on. Can you please look at the `handbook.tex` file, find the text "Before installing musl Void", and tell me what appears in the first set of braces for the `\hyperlink` command in that sentence, i.e. the link name? On my system, the link name is very clearly broken.

**Comment by flexibeast** [comment]
Gah. The problem is that a line of code in `mdbook-latex` assumes that the repo directory is called `void-docs`, so things didn't work when i did a fresh clone into a directory called `void-docs-2`. Sorry. :-(

Merging this PR.

**Comment by ericonr** [comment]
Damn, I see. No worries, glad you found it! Btw, if you want to avoid cloning repos all the time, I'm using https://github.com/leahneukirchen/git-merge-pr to check out PRs locally.

**Pull request merged**